test/test_crypto.py

```diff
@@ -33,53 +33,36 @@ class TestEncryptUtils(unittest.TestCase):
     
     def setUp(self):
         '''
-        Some of the functions require the certificate and key to be files.
+        If no key/cert pair found, generate a new
+        key/cert pair and store as a file.
         '''
-        self.certfile, self.certpath = tempfile.mkstemp(prefix='cert')
-        os.write(self.certfile, TEST_CERT)
-        os.close(self.certfile)
+        if not (os.path.exists(TEST_CERT_FILE) and os.path.exists(TEST_KEY_FILE)):
+            # create a key/cert pair
+            call(['openssl', 'req', '-x509', '-nodes', '-days', '365',
+                  '-newkey', 'rsa:2048', '-keyout', TEST_KEY_FILE,
+                  '-out', TEST_CERT_FILE, '-subj', TEST_CERT_DN])
 
-        self.keyfile, self.keypath = tempfile.mkstemp(prefix='key')
-        os.write(self.keyfile, TEST_KEY)
-        os.close(self.keyfile)
-        
-        # Set up an openssl-style CA directory, containing the 
-        # self-signed certificate as its own CA certificate, but with its
-        # name as <hash-of-subject-DN>.0.
-        
-        ca_certs = [TEST_CERT]
-
-        self.ca_dir = tempfile.mkdtemp(prefix='ca')
-        for cert in ca_certs:
+            # Set up an openssl-style CA directory, containing the 
+            # self-signed certificate as its own CA certificate, but with its
+            # name as <hash-of-subject-DN>.0.
             p1 = Popen(['openssl', 'x509', '-subject_hash', '-noout'],
                        stdin=PIPE, stdout=PIPE, stderr=PIPE)
-            hash_name, unused_error = p1.communicate(cert)
-            ca_certpath = os.path.join(self.ca_dir, hash_name.strip() + '.0')
-            ca_cert = open(ca_certpath, 'w')
-            ca_cert.write(cert)
-            ca_cert.close()
+
+            with file(TEST_CERT_FILE) as test_cert:
+                cert_string = test_cert.read()
```

Comment:
You've finished using `test_cert` by this stage, so should dedent the following lines.